src/EnergyPlus/CommandLineInterface.cc

```diff
@@ -188,15 +188,27 @@ namespace CommandLineInterface {
 
         state.dataGlobal->outputEpJSONConversionOnly = opt.isSet("--convert-only");
 
+        bool eplusRunningViaAPI = state.dataGlobal->eplusRunningViaAPI;
+
         // Process standard arguments
         if (opt.isSet("-h")) {
             DisplayString(state, usage);
-            exit(EXIT_SUCCESS);
+            if (eplusRunningViaAPI) {
+                // we need another return code to let runEnergyPlusAsLibrary know it should not try to run anything
+                return static_cast<int>(ReturnCodes::SuccessButHelper);
```

Comment:
Added a third return code, in case it "works" but you still do not want to try and run the simulation...